### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by greypants**
datepicker

More to do, but this is the basic implementation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "license": "MIT",
+  "devDependencies": {
+    "babel-core": "^6.3.15",
+    "babel-loader": "^6.2.0",
+    "babel-preset-es2015": "^6.3.13",
+    "browser-sync": "^2.10.1",
+    "webpack": "^1.12.9"
+  },
+  "dependencies": {
+    "delegate": "^3.0.0"
```
Comment (nhunzaker) [inline package.json:29]: what does `delegate` do? should this be peer or `~3.0.0`?

Comment (nhunzaker) [inline package.json:29]: Sorry. I'll look up delegate on npm. semver question still stands

File: src/DatePicker/index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import delegate from 'delegate'
+import defaults from './lib/defaults'
+import zeroPad from './lib/zeroPad'
+
+export default class DatePicker {
+  constructor(el, options = {}) {
+    this.constants = Object.assign({}, defaults, options)
```
Comment (nhunzaker) [inline src/DatePicker/index.js:7]: Object.assign is still pretty new. What about pulling in https://github.com/babel/babel/tree/development/packages/babel-plugin-transform-object-assign ?

**Comment by nhunzaker** [comment]
Pretty cool! :+1:

File: src/DatePicker/index.js (hunk 2 above)
Comment (greypants) [inline src/DatePicker/index.js:7]: oops. Good call!

File: package.json (hunk 1 above)
Comment (greypants) [inline package.json:29]: it's a dom event delegation helper like `jQuery.on`

Comment (greypants) [inline package.json:29]: will change to peerDependency.

**Pull request merged**